Sources/Security/HTTPRequest+Signing.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  HTTPRequest+Signing.swift
+//
+//  Created by Nacho Soto on 11/16/23.
+
+import CryptoKit
+import Foundation
+
+extension HTTPRequest {
+
+    static func headerParametersForSignatureHeader(headers: Headers) -> String? {
+        guard #available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *) else {
+            // Signature verification is not available.
+            return nil
+        }
+
+        let headersToSign = Self.headersToSign.map(\.rawValue)
+
+        if !headersToSign.isEmpty, let hash = Self.postParameterHash(headers) {
+            return Self.signatureHashHeader(keys: headersToSign, hash: hash)
+        } else {
+            return nil
+        }
+    }
+
+    /// - Returns: `nil` if none of the requested headers are found
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    private static func postParameterHash(_ headers: Headers) -> String? {
+        let headersToSign = Self.headersToSign.map(\.rawValue)
+        let values = headers
+            .filter { headersToSign.contains($0.key) }
+            .map(\.value)
```

Comment:
Will this respect the order? we need values to be in the order of headers to sign, don't we?

Reply:
Good eye! This is backwards, it needs to be derived from `headersToSign` which is ordered (`headers` is a `Map` so it's not).
It didn't matter right now because we only sign one header, but it would have been non-deterministic as soon as we added a second one, so thanks! I just fixed it :)